linux/DtaDevLinuxSata.cpp

```diff
@@ -87,6 +88,86 @@ bool DtaDevLinuxSata::init(const char * devref)
 	return isOpen;
 }
 
+/*
+ * Determines if the transport for the given file descriptor `fd` is SATA.
+ *
+ * Steps:
+ * 1. Resolve `/proc/self/fd/<fd>` to get the SCSI device name.
+ * 2. Read the symlink `/sys/class/block/<sdX>/device` to extract
+ * the Host Bus Number.
+ * 3. Check `/sys/class/scsi_host/<Host>/proc_name` to see if the driver
+ * is "ahci".
+ *
+ * Returns true if the transport is SATA; false otherwise.
+ */
+bool is_sata_transport(int fd)
+{
+    char dev_path[1024];
+    char host_link[1024];
+    try {
+        string fd_path = "/proc/self/fd/" + to_string(fd);
+        ssize_t len = readlink(fd_path.c_str(), dev_path, sizeof(dev_path) - 1);
+        if (len == -1)
+            return (false);
+        dev_path[len] = '\0';
+        string dev_name = string(dev_path);
+        if (dev_name.substr(0, 5) != "/dev/")
+            return (false);
+        dev_name = dev_name.substr(5);
+        string host_path = "/sys/class/block/" + dev_name + "/device";
+        len = readlink(host_path.c_str(), host_link, sizeof(host_link) - 1);
+        if (len == -1)
+            return (false);
+        host_link[len] = '\0';
+        string host_str = string(host_link);
+        size_t pos = host_str.find_last_of('/');
+        if (pos == string::npos)
+            return (false);
+        size_t colon_pos = host_str.find(':', pos);
+        if (colon_pos == string::npos)
+            return (false);
+        int host_id = stoi(host_str.substr(pos + 1, colon_pos - pos - 1));
+        ifstream proc_file("/sys/class/scsi_host/host" + to_string(host_id) + "/proc_name");
+        string proc_name;
+        if (getline(proc_file, proc_name) && proc_name == "ahci")
```

Comment:
AHCI is not the only ATA transport available.  Any libata driver should go that way.

Reply:
Indeed, updated as [done by lsblk](https://github.com/util-linux/util-linux/blob/master/misc-utils/lsblk.c#L547).